src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -122,57 +122,66 @@ const StudentApplication = ({ userProfile, authentication }) => {
   const [snackbarSeverity, setSnackbarSeverity] = useState('');
   const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
 
-  // const peopleSearchRef = useRef();
-
   // Off-campus program info is stored as a Map, where the Key is a student's username and the corresponding Value is the department of that student's off-campus program
   let offCampusProgramInfo = new Map();
   applicants.forEach((profile) => offCampusProgramInfo.set(profile.AD_Username, ''));
 
-  /**
-   * Attempt to load an existing application from the database if one exists
-   */
-  const loadSavedApplication = useCallback(async () => {
-    // TODO: Implement this once save/load of application data has been implemented in the backend
-    setLoading(true);
-    // Check if the current user is on an application. Returns the application ID number if found
-    let newApplicationID = -1; // await housing.getApplicationID();
-    if (newApplicationID !== null && newApplicationID !== -1) {
-      setApplicationID(newApplicationID);
-      let applicationDetails = await housing.getApartmentApplication(newApplicationID);
-      if (applicationDetails) {
-        if (applicationDetails.DateSubmitted) {
-          setDateSubmitted(applicationDetails.DateSubmitted);
-        }
-        if (applicationDetails.DateModified) {
-          setDateModified(applicationDetails.DateModified);
+  useEffect(() => {
+    /**
+     * Loads the user's saved apartment application, if one exists
+     */
+    const loadSavedApplication = async () => {
+      // TODO: Implement this once save/load of application data has been implemented in the backend
+      setLoading(true);
+      // Check if the current user is on an application. Returns the application ID number if found
+      let newApplicationID = null;
+      try {
+        newApplicationID = await housing.getApplicationID();
+        console.log('Retrieved Application ID: ' + newApplicationID);
+        if (newApplicationID === null || newApplicationID === -1) {
+          // Intentionally trigger the 'catch'
+          throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
```

Comment:
Is it necessary to have this error-checking/debugging merged?

Reply:
This is not a debugging feature. As the comment above it says, the error also serves the purpose of forcing the `try` to go to its `catch`, which will make sure the variables are set to the correct default values. 
It is important that the code exits the `try` section if the `newApplicationID` is an invalid value so that it does not waste time calling the "load" API endpoint with a nonsensical ID value